csrc/multidevice/communicator.cpp

```diff
@@ -196,6 +197,8 @@ Communicator::Communicator(
     return;
   }
 
+  NVFUSER_CUDA_RT_SAFE_CALL(cudaSetDevice(local_rank_));
```

Comment:
It's been a long time I suspected this was going to be required at some point. Anyway, this is a recommended (if not required) practice. Without it, `cudaEventRecord` throws `cudaErrorInvalidResourceHandle` in a multi-GPU scenario.